go/app/main.go

```diff
@@ -48,7 +48,22 @@ func root(c echo.Context) error {
 	return c.JSON(http.StatusOK, res)
 }
 
-var itemlist ItemList
+func decodeJson() (*ItemList, error) {
+	jsonfile, err := os.Open("items.json")
+	if err != nil {
+		return nil, err
+	}
+	defer jsonfile.Close()
+
+	//decode
+	var itemlist ItemList
+	decoder := json.NewDecoder(jsonfile)
+	if err := decoder.Decode(&itemlist); err != nil {
+		return nil, err
+	}
+
+	return &itemlist, nil
+}
```

Comment:
[nit] mainコードの可読性を考えるとポインターにしなくてもいいような気がします

```suggestion
func decodeJson() (ItemList, error) {
	jsonfile, err := os.Open("items.json")
	if err != nil {
		return nil, err
	}
	defer jsonfile.Close()

	//decode
	var itemlist ItemList
	decoder := json.NewDecoder(jsonfile)
	if err := decoder.Decode(&itemlist); err != nil {
		return nil, err
	}

	return itemlist, nil
}
```

Reply:
返り値の型をItemlistにするとreturn nilでエラーが起きたのでポインタにしました

Reply:
return nilで返すのではなく、空のlistを返すようにしてもいいかもしれないです。とりあえずこの部分はstep4ではつかわないのでそのままでも大丈夫です。